### PR TITLE
vp9dec: create surface and context with ivf width and height.

### PR DESCRIPTION
**Pull request by zhongcong**
Signed-off-by: Zhong Cong <congx.zhong@intel.com>

### CODE REVIEW

File: decoder/vaapidecoder_vp9.cpp [hunk 1]
```diff
@@ -90,18 +90,26 @@ void VaapiDecoderVP9::flush(void)
 Decode_Status VaapiDecoderVP9::ensureContext(const Vp9FrameHdr* hdr)
 {
     // only reset va context when there is a larger frame
-    if (m_configBuffer.width < hdr->width
```
Comment (xuguangxin) [inline decoder/vaapidecoder_vp9.cpp]: why we need change so many code, can we just call VaapiDecoderBase::start in vp9 decoder's start?

File: tests/decodeinput.cpp [hunk 2]
```diff
@@ -45,6 +45,8 @@ class MyDecodeInput : public DecodeInput{
     virtual bool isEOS() {return m_parseToEOS;}
     virtual bool init() = 0;
     virtual const string& getCodecData();
+    uint16_t getWidth() {return 0;}
```
Comment (xuguangxin) [inline tests/decodeinput.cpp]: why we not return m_width and m_height here?

File: tests/decodeinput.cpp [hunk 3]
```diff
@@ -247,6 +260,16 @@ bool DecodeInputVPX::getNextDecodeUnit(VideoDecodeBuffer &inputBuffer)
     return true;
 }
 
+uint16_t DecodeInputVPX::getWidth()
+{
+    return m_width;
```
Comment (xuguangxin) [inline tests/decodeinput.cpp]: if we return m_width m_height in base class, this code is not needed.

File: tests/decodeinputavformat.h [hunk 4]
```diff
@@ -44,6 +44,8 @@ class DecodeInputAvFormat : public DecodeInput
     virtual const char * getMimeType();
     virtual bool getNextDecodeUnit(VideoDecodeBuffer &inputBuffer);
     virtual const string& getCodecData();
+    uint16_t getWidth() {return 0;}
```
Comment (xuguangxin) [inline tests/decodeinputavformat.h]: did  you tried use avformat to decode ivf file? Maybe we need return some width & height here too

**Comment by zhongcong** [comment]
updated.

File: tests/decodeinput.cpp [hunk 5]
```diff
@@ -45,11 +45,15 @@ class MyDecodeInput : public DecodeInput{
     virtual bool isEOS() {return m_parseToEOS;}
     virtual bool init() = 0;
     virtual const string& getCodecData();
+    uint16_t getWidth() {return m_width;}
```
Comment (xuguangxin) [inline tests/decodeinput.cpp]: could we move this two functions to DecodeInput and remove similar functions in DecodeInputAvFormat

File: decoder/vaapidecoder_base.h [hunk 6]
```diff
@@ -92,7 +92,6 @@ class VaapiDecoderBase:public IVideoDecoder {
     VideoConfigBuffer m_configBuffer;
     VideoFormatInfo m_videoFormatInfo;
 
-
```
Comment (xuguangxin) [inline decoder/vaapidecoder_base.h]: why we need change this?

File: tests/decodeinputavformat.cpp [hunk 7]
```diff
@@ -29,7 +29,7 @@
 #include "interface/VideoDecoderDefs.h"
 
 DecodeInputAvFormat::DecodeInputAvFormat()
-:m_format(NULL),m_videoId(-1), m_codecId(AV_CODEC_ID_NONE), m_isEos(true)
+:m_format(NULL),m_videoId(-1), m_codecId(AV_CODEC_ID_NONE), m_isEos(true), m_width(0), m_height(0)
```
Comment (xuguangxin) [inline tests/decodeinputavformat.cpp]: this line is not needed if you move m_width && m_height to DecodeInput

**Comment by xuguangxin** [comment]
+1

**Comment by xuguangxin** [comment]
merged as a4cb390bec76d260e5af24b3ff6a5feeaf022938

**Pull request closed without merging**